main.cpp

```diff
@@ -38,7 +38,7 @@ int main(int argc, char *argv[])
     // For simplicity, convert all arguments to strings
     std::vector<std::string> args;
     for (int i = 0; i < argc; i++) {
-        args.push_back(argv[i] | filter::lowercase);
+        args.push_back(argv[i]);
```

Comment:
why this unrelated change?

Reply:
Because t was wrong. For example, my code is checked out in a directory named Atlantis, and lower-casing the full path to the intro.html file will cause it to not be found by the genrules command.

Reply:
Hmmm yeah.. a better choice there would be to use `std::vector<strings::ci_string>` which preserves the original case of the input, but compares arguments case insensitively, which mirrors the original AString behavior (this change here predates my introduction of that ci_string class for those specific cases which needed it).. I am ok with this, and will probably just change it to use ci_string sometime in the (near) future unless you want to do it as part of this?

Reply:
strings::ci_string doesn't convert to lowercase, just applies case-insenstive comparators.  As I said, this was (as you pointed out) a broken attempt to retain the original AString comparison behavior when I removed AString from the file and argument handling initially.  This code *used* to convert all arguments to AStrings. strings::ci_string was the cleaned up version of that but I forgot to fix it here.  Either way, I am willing to accept this piece as is.